src/m_class.c

```diff
@@ -569,28 +569,27 @@ void class_addcreator(t_newmethod newmethod, t_symbol *s,
     t_atomtype type1, ...)
 {
     va_list ap;
-    t_atomtype vec[MAXPDARG+1], *vp = vec;
+    t_atomtype vec[MAXPDARG+1], *vp = vec, argtype = type1;
     int count = 0;
-    *vp = type1;
+    if(!argtype) argtype = A_NULL;
```

Comment:
What is this supposed to do? `A_NULL` is defined as 0, so this line seems entirely redundant.

Reply:
a well, it is not *defined* as `0`. it just happens to be the first enum in the list (thus it is *implicitly* `0`).
the idea of this line (and the one below) is to make this explicit.

i figure any optimizing compiler would detect the same-ness and eliminate the redundant call.

Reply:
> it just happens to be the first enum in the list

That means it *is* defined as 0, you don't need to spell it out.

> the idea of this line (and the one below) is to make this explicit.

But why? In the while loop you check for `argtype != A_NULL`, that is all we are interested in. We don't even need to care about the numeric value of `A_NULL`.

> i figure any optimizing compiler would detect the same-ness and eliminate the redundant call.

I'm not worried about the optimizer :) This line just caught me off-guard and I left me scratching my head for a while.

I guess you wanted to say "instead of `A_NULL` you can pass `0`. which are really the same thing". In that case, I would prefer a simple comment :)

Reply:
indeed, that would be the better solution.
i thought about it, but somehow did not want to change any headers in the commit.